R/Data-methods.R

```diff
@@ -302,11 +301,15 @@ setMethod(
     object@ID <- c(object@ID, safeInteger(ID))
 
     # Add cohort number.
-    last_cohort <- max(tail(object@cohort, 1L), 0L)
-    object@cohort <- c(
-      object@cohort,
-      rep(last_cohort, y_len) + ifelse(new_cohort, 1L, 0L)
-    )
+    if (object@nObs == 0) {
+      new_cohort_ID <- 1L
+    } else {
+      new_cohort_ID <- tail(object@cohort, 1L) + ifelse(new_cohort, 1L, 0L)
+    }
```

Comment:
```suggestion
    new_cohort_id <- if (object@nObs == 0) {
      1L
    } else {
      tail(object@cohort, 1L) + ifelse(new_cohort, 1L, 0L)
    }
```
(generally better in this kind of patterns) plus avoid upper case in new internal variable names (we keep them when they are already there for external interface to avoid code breaking)